src/api/grainConfig.js

```diff
@@ -30,56 +28,24 @@ export const parser: C.Parser<GrainConfig> = C.object(
  * fields can form valid policies.
  */
 export function toDistributionPolicy(x: GrainConfig): DistributionPolicy {
-  const immediatePerWeek = NullUtil.orElse(x.immediatePerWeek, 0);
-  const recentPerWeek = NullUtil.orElse(x.recentPerWeek, 0);
-  const balancedPerWeek = NullUtil.orElse(x.balancedPerWeek, 0);
-
-  if (!isNonnegativeInteger(immediatePerWeek)) {
-    throw new Error(
-      `immediate budget must be nonnegative integer, got ${immediatePerWeek}`
-    );
-  }
-  if (!isNonnegativeInteger(recentPerWeek)) {
-    throw new Error(
-      `recent budget must be nonnegative integer, got ${recentPerWeek}`
-    );
+  if (!x.allocationPolicies.length) {
+    throw new Error(`Please provide at least one allocation policy, got zero.`);
```

Comment:
Zero is a valid input with an unambiguous interpretation. Throwing errors on such inputs just makes the system harder to reason about. Please support zero.

Reply:
Done.